### ISSUE TITLE
osu!lazer black screen

### ISSUE THREAD

**Issue by kli75962** [ppy/osu#17552]
### Type

Crash to desktop

### Bug description

osu!lazer will black screen sometimes, need to ctrl f4 to reopen it

### Version

2022.327.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/8377420/database.log)
[network.log](https://github.com/ppy/osu/files/8377421/network.log)
[performance.log](https://github.com/ppy/osu/files/8377422/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8377423/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8377424/updater.log)

**Comment by peppy** [comment]
Can you provide more details on your operating environment (platform, gpu) and also whether this only happens in fullscreen mode?

**Comment by kli75962** [comment]
win10 on desktop computer, NVIDIA Geforce GTX 1660 SUPER, I only play osu on fullscreen since the input latency so I don't know if it happens on another screen mode

**Comment by kli75962** [comment]
it only happens on the menu, in-game is totally fine

**Comment by peppy** [comment]
When it happens do sounds from the game still play as usual (ie. if you hit escape or arrow keys, can you hear that the game is still functioning)?

As a first step, can you confirm that your graphics card drivers are up-to-date?

**Comment by kli75962** [comment]
yes the game is still functioning, I haven't updated to the newest version on 22/3 yet

**Comment by kli75962** [comment]
I will update now

**Comment by andy840119** [comment]
![image](https://user-images.githubusercontent.com/9100368/161051856-e337e159-0bbc-425a-bdc4-1d18d61600b4.png)

It happened on my PC also(windows 11 with R7-2700X and R9-280X graphic card).
But seems it only happened in the release build.

Also, seems it only happened in the release build(for more than two weeks I remember).
I cannot make the screen blank in the debug build of Lazer.

Also, the blank screen never happened while developing my ruleset.
But it usually happened in the released Lazer on my PC.

**Comment by smoogipoo** [comment]
Does resizing the window fix it?

**Comment by andy840119** [comment]
Resize windows not fix it.
I can try if change refresh rate or change to full-screen can fix that after 3 or 4 days

**Comment by pscr00ge** [comment]
updating, ive got black backgrounds on Default backgrounds (ugly, not season bgs) and on song selection screen. the most interesting part is that i was still able to see ui and still was able to move around in menu and play with ultra lag spikes if i didnt hit circles sliders spinners in session before

platform:android

**Comment by pscr00ge** [comment]
[database.log](https://github.com/ppy/osu/files/8403896/database.log)
[network.log](https://github.com/ppy/osu/files/8403897/network.log)
[performance.log](https://github.com/ppy/osu/files/8403898/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8403899/runtime.log)
logs if you want eat them, dunno if these logs contains these glitches

**Comment by smoogipoo** [comment]
@pscr00ge Your issue's different because you're on Android, and I have a suspicion you are suspending/backgrounding the app, is this the case?

```
2022-04-02 20:02:13 [error]: osuTK.Graphics.GraphicsContextException: Failed to swap buffers for context 1869378080 current. Error: BAD_SURFACE
2022-04-02 20:02:13 [error]: at osuTK.Platform.Egl.EglContext.SwapBuffers () [0x00041] in <d14e0ddeba9c4ce7aaaa9213c3d63316>:0
2022-04-02 20:02:13 [error]: at osuTK.Graphics.GraphicsContext.SwapBuffers () [0x00000] in <d14e0ddeba9c4ce7aaaa9213c3d63316>:0
2022-04-02 20:02:13 [error]: at osuTK.Android.AndroidGameView.SwapBuffers () [0x0001a] in <ad754a1d77a04e1aa2d11b9bd1285774>:0
2022-04-02 20:02:13 [error]: at osu.Framework.Platform.OsuTKWindow.SwapBuffers () [0x00000] in <ea743036e3ab473386cdb4067bcb5a26>:0
2022-04-02 20:02:13 [error]: at osu.Framework.Platform.GameHost.Swap () [0x00006] in <ea743036e3ab473386cdb4067bcb5a26>:0
2022-04-02 20:02:13 [error]: at osu.Framework.Platform.GameHost.DrawFrame () [0x00171] in <ea743036e3ab473386cdb4067bcb5a26>:0
2022-04-02 20:02:13 [error]: at osu.Framework.Threading.GameThread.processFrame () [0x000b9] in <ea743036e3ab473386cdb4067bcb5a26>:0
```

**Comment by cow03haha** [comment]
i got this problem too, black screen but game still running, i have try resize window but not work
platform: windows 11
cpu: i7-9750h
gpu: nvidia gtx 1650

[network.log](https://github.com/ppy/osu/files/8707277/network.log)
[performance.log](https://github.com/ppy/osu/files/8707278/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8707279/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8707280/updater.log)
[database.log](https://github.com/ppy/osu/files/8707281/database.log)

**Comment by bdach** [comment]
NVIDIA card owners reporting this should start by (a) checking device driver updates, and (b) checking if disabling shadowplay does anything.

**Comment by cow03haha** [comment]
> NVIDIA card owners reporting this should start by (a) checking device driver updates, and (b) checking if disabling shadowplay does anything.

i try to update driver and disable nvidi in-game overlay, still got black screen
btw, I observe that osu will black screen during browse beatmap listing(ctrl + d)

![圖片](https://user-images.githubusercontent.com/44705326/169942484-e9aab403-e856-41bc-bded-af6aa1708918.png)

**Comment by appleneko2001** [comment]
Hi everyone!
I have same problem that might related with this. I recorded my screen how it happen (not accurate reproduce because it doesn't often appears). https://youtu.be/Zhks_xIsweA
My PC specs:
Graphics card: NVIDIA GeForce GTX 1050Ti Driver Ver. 512.59
Processor: AMD Ryzen 5 3600

Logging: [logs.zip](https://github.com/ppy/osu/files/8766891/logs.zip)
PS: sometimes the game will crash with error window but sometimes just got black screen.

**Comment by appleneko2001** [comment]
> Hi everyone! I have same problem that might related with this. I recorded my screen how it happen (not accurate reproduce because it doesn't often appears). https://youtu.be/Zhks_xIsweA My PC specs: Graphics card: NVIDIA GeForce GTX 1050Ti Driver Ver. 512.59 Processor: AMD Ryzen 5 3600
> 
> Game ver. 2022.523.0
> 
> Logging: [logs.zip](https://github.com/ppy/osu/files/8766891/logs.zip) PS: sometimes the game will crash with error window but sometimes just got black screen.

It wont crash anymore after I change multi thread to single thread looks like

**Comment by UknTatsu** [comment]
Hi guys, not sure if this will help, but I have the same problem of osu!lazer showing just a black screen.
I did find out though that opening the normal version of osu! before launching osu!lazer seems to make it work. However, closing normal osu! will make osu!lazer crash.

**Comment by k3-cat** [comment]
I got the exactly same issue with Windows 11 (build 22623.1037)
cpu: i7-11800H
gpu: RTX-3060 (laptop)

logs show no abnormal signs.

**Comment by k3-cat** [comment]
I think this problem is due to multithread (as @appleneko2001's method do works).

Multithread always cause some extra problems, and sadly currently there seems has no successful sign that prove multithread UI works.

**Comment by appleneko2001** [comment]
But I don't think multithread should be removed, there have a good reason to use multithreading. 
I thought ppy want to make osu stable running even framerate are pretty low, or multicore optimizations. 

Anyway the workaround for this scenario is use single thread, but its not a solution (my opinion).
Interesting point is recently my game wont black screen anymore while using multithreading (latest version, 2022.1228.0).

**Comment by peppy** [comment]
Please wait for the new renderer which will likely fix this.

**Comment by peppy** [comment]
Closing as should-be-resolved.

Please feel free to reply in this thread if you can still reproduce an issue.
